From 978ede6dcba6a86156e9332cf2cc74236c4b7900 Mon Sep 17 00:00:00 2001
From: Breno Lima <breno.lima@nxp.com>
Date: Mon, 26 Aug 2019 18:05:05 -0300
Subject: [PATCH 26/31] MLK-22505 mx7ulp: Enable UUU support for
 mx7ulp_evk_m4boot_defconfig target

Currently it's not possible to boot mx7ulp_evk_m4boot_defconfig target:

  OBJCOPY examples/standalone/hello_world.bin
  LD      u-boot
arch/arm/lib/built-in.o: In function `setup_serial_tag':
/home/breno/NXP/bootloader/uboot-imx/arch/arm/lib/bootm.c:192: \
undefined reference to `get_board_serial'
Makefile:1505: recipe for target 'u-boot' failed
make: *** [u-boot] Error 1

Fix this issue by enabling UUU support for mx7ulp_evk_m4boot_defconfig target.

Reviewed-by: Ye Li <ye.li@nxp.com>
Signed-off-by: Breno Lima <breno.lima@nxp.com>
(cherry picked from commit 7fab9bf1bbcaa93f91cd69c970e92676011a47d3)
---
 configs/mx7ulp_evk_m4boot_defconfig | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/configs/mx7ulp_evk_m4boot_defconfig b/configs/mx7ulp_evk_m4boot_defconfig
index 6b1135c..da8170f 100644
--- a/configs/mx7ulp_evk_m4boot_defconfig
+++ b/configs/mx7ulp_evk_m4boot_defconfig
@@ -59,3 +59,18 @@ CONFIG_USB_ETHER_RTL8152=y
 CONFIG_VIDEO=y
 CONFIG_IMX_NORTHWEST_MIPI_DSI=y
 CONFIG_IMX_M4_BIND=y
+CONFIG_USB_GADGET=y
+CONFIG_USB_GADGET_DOWNLOAD=y
+CONFIG_USB_GADGET_MANUFACTURER="FSL"
+CONFIG_USB_GADGET_VENDOR_NUM=0x0525
+CONFIG_USB_GADGET_PRODUCT_NUM=0xa4a5
+CONFIG_CI_UDC=y
+
+CONFIG_CMD_FASTBOOT=y
+CONFIG_USB_FUNCTION_FASTBOOT=y
+CONFIG_FASTBOOT_UUU_SUPPORT=y
+CONFIG_FASTBOOT=y
+CONFIG_FASTBOOT_BUF_ADDR=0x60800000
+CONFIG_FASTBOOT_BUF_SIZE=0x20000000
+CONFIG_FASTBOOT_FLASH=y
+CONFIG_EFI_PARTITION=y
-- 
2.7.4

